### DISCUSSION TITLE
Some ideas for improvements to the fetchart plugin.

### DISCUSSION THREAD

**Discussion by giorgosg** [beetbox/beets#4885] [Ideas]
It currently does not check the filesystem when running on already imported albums. I have many covers downloaded with other programs in the album folder when they where not found so it would be nice to be able to get them into the beets database.
Also a nice feature would be to check the current cover against the current minwidth/maxwidth and search again if it doesn't fit it.

**Comment by jackwilsdon** [comment]
>It currently does not check the filesystem when running on already imported albums.

If the `filesystem` source is enabled, running `beet fetchart -f [search term]` should use the artwork from the album's folder (assuming it's named correctly). Is this not working for you?

**Reply by giorgosg** [reply]
It doesn't seem to be working. It works correctly and uses the file from the filesystem when importing, but when using beets fetchart it doesn't use the file. It sometimes finds a cover and names it cover.1.jpg when there is a good cover.jpg

That is with the latest release I will try with the git version soon.

**Reply by giorgosg** [reply]
Ok... It looks like it finds the file from the filesystem, but instead of using it as it is it copies it to cover.1.jpg.
